app/models/user.rb

```diff
@@ -4,9 +4,9 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable, :confirmable
 
-  has_many :posts, foreign_key: 'author_id'
-  has_many :comments, foreign_key: 'author_id'
-  has_many :likes, foreign_key: 'author_id'
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes, foreign_key: 'author_id', dependent: :destroy
 
   validates :name, presence: true, length: { in: 3..20 }
   validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```

Comment:
- [OPTIONAL] You can also create your `user.admin?` method inside the user model. 👇🏻 

```ruby
...
   def admin?
     role == 'admin' ? true : false
   end
```

Reply:
Hey, @BenMukebo. Thanks for the feedback and review, Implemented the `user.admin?` and used the ability model to check the role.